pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1048,6 +1082,57 @@ func (m *ManagerImpl) GetDevices(podUID, containerName string) ResourceDeviceIns
 	return m.podDevices.getContainerDevices(podUID, containerName)
 }
 
+func (m *ManagerImpl) UpdateAllocatedResourcesStatus(pod *v1.Pod, status *v1.PodStatus) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	// Today we ignore edge cases that are not likely to happen:
+	//  - update statuses for containers that are in spec, but not in status
+	//  - update statuses for resources requested in spec, but with no information in podDevices
+	for i, containerStatus := range status.ContainerStatuses {
+		devices := m.podDevices.getContainerDevices(string(pod.UID), containerStatus.Name)
+
+		for resourceName, deviceInstances := range devices {
+			for id, d := range deviceInstances {
+				health := pluginapi.Healthy
+				// this is unlikely, but check for existence here anyways
+				if r, ok := m.allDevices[resourceName]; ok {
+					if _, ok := r[id]; ok {
+						health = m.allDevices[resourceName][id].Health
+					}
+				}
+
+				d.Health = health
+
+				deviceInstances[id] = d
```

Comment:
this suggests some better abstraction/support in device manager or pod_devices can help. Let's evaluate in the beta scope. I can't see obvious issues besides the concurrency (already addressed) atm, so we can keep it like it is for alpha

Reply:
ok, yes, we have a lot of data structures that needs to be kept in sync today. Looks like they grew slowly over time and a refactoring would be beneficial. This works for alpha, but when we will be implementing the "Unknown" status - a lot of changes will be needed. And it will be a good opportunity to refactor. This was one of the main reason to not implement the Unknown in this iteration.